From 9853bdb0b04ec8cd28483b127699ebd9cb305a77 Mon Sep 17 00:00:00 2001
From: Xiretza <xiretza@xiretza.xyz>
Date: Sat, 4 Dec 2021 19:47:58 +0100
Subject: [PATCH 2/2] compilers.xml: use gcc --version to get version number
 instead of gcc -v

The output of gcc -v is dependent on locale can thus break the regex.

See also https://github.com/AdaCore/gprconfig_kb/issues/10
---
 db/compilers.xml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/db/compilers.xml b/db/compilers.xml
index ac9a701..416329a 100644
--- a/db/compilers.xml
+++ b/db/compilers.xml
@@ -22,9 +22,9 @@
     <name>GCC</name>
     <executable prefix="1">(.*-wrs-.*|.*-sysgo.*|.*-elf-.*|.*-eabi-.*|.*-eabispe-.*|avr-.*|.*-elinos-linux.*|.*-generic-linux-gnu-|.*mingw32.*|.*-xcoff-.*|arm-linux-androideabi-|arm-linux-gnueabi-|arm-linux-gnueabihf-|e500v2-gnu-linux-|powerpc-.*-linux.*-|.*-darwin.*-|aarch64-linux-gnu-)?gcc</executable>
     <version>
-      <external>${PREFIX}gcc -v</external>
+      <external>${PREFIX}gcc --version</external>
       <nogrep regexp="for GNAT Pro"></nogrep>
-      <grep regexp="^gcc \S+ (\S+)" group="1"></grep>
+      <grep regexp="^gcc \(GCC\) (\S+)" group="1"></grep>
     </version>
     <languages>C</languages>
     <target>
-- 
2.34.1

